jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter.java

```diff
@@ -174,30 +176,25 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 			Map<String, Object> resultValues = result.getValues();
 			if (resultValues != null) {
 				for (Entry<String, Object> values : resultValues.entrySet()) {
-					String line = null;
-
-					if (NumberUtils.isNumeric(values.getValue())) {
-						line = KeyUtils.getKeyString(server, query, result, values, typeNames, rootPrefix)
-								+ ":" + values.getValue().toString() + "|" + bucketType + "\n";
-					} else if (stringsValuesAsKey) {
-						line = KeyUtils.getKeyString(server, query, result, values, typeNames, rootPrefix)
-								+ "." + values.getValue().toString() + ":" + stringValueDefaultCount.toString()
-								+ "|" + bucketType + "\n";
-					}
-
-					if (StringUtils.isNotBlank(line)) {
-						doSend(line.trim());
-					}
+					if (!isNumeric(values.getValue()) && !!stringsValuesAsKey) break;
+
+					String line = KeyUtils.getKeyString(server, query, result, values, typeNames, rootPrefix)
+								+ ":" + computeActualValue(values) + "|" + bucketType + "\n";
+
+					doSend(line.trim());
 				}
 			}
 		}
 	}
 
+	private String computeActualValue(Entry<String, Object> values) {
+		if (isNumeric(values.getValue())) return values.getValue().toString();
+		return stringValueDefaultCount.toString();
```

Comment:
This computation is not valid, because the ``values.getValue().toString()`` should in the key sent to statsd, before the ``: ``.